edgedb/asyncio_con.py

```diff
@@ -155,6 +166,13 @@ async def aclose(self):
             try:
                 self._protocol.terminate()
                 await self._protocol.wait_for_disconnect()
+
+                # With asyncio on CPython 3.10 or lower, a normal exit
+                # (connection closed by peer) cannot set the transport._closed
+                # properly, leading to false ResourceWarning. Let's fix that by
+                # closing the transport again.
+                if not self._transport.is_closing():
+                    self._transport.close()
```

Comment:
Should this be copied to `terminate()`?

Reply:
I think `_AsyncIOConnectionImpl.terminate()` is fine, because it calls `AsyncIOProtocol.abort()` which closes the transport.

Reply:
But maybe moving this logic to `AsyncIOProtocol.connection_lost()` is a better idea - so that the transport is always marked as closed as far as `connection_lost()` is called.